src/main/java/helper/SuccessResponseGenerator.java

```diff
@@ -108,38 +122,50 @@
         return responseBody;
     }
 
-    private static Map<String, Object> getMessageForResponseBody(long userId, Message message) {
-    
-        Map<String, Object> messageForResponseBody = new LinkedHashMap<String, Object>();
+    private static Map<String, List<Map<String, Object>>> getResponseBodyForListMessages(List<Map<String, Object>> listOfMessages) {
+        Map<String, List<Map<String, Object>>> responseBody = new LinkedHashMap<String, List<Map<String, Object>>>();
+        responseBody.put("payload", listOfMessages);
+       
+        return responseBody;
+    }
 
-        messageForResponseBody.put("MessageId", message.getMessageId());
-        messageForResponseBody.put("CreationTs", message.getCreationTs());
-        messageForResponseBody.put("ChatId", message.getChatId());
-        messageForResponseBody.put("SentByCurrentUser", message.getSenderId() == userId);
-        messageForResponseBody.put("TextContent", message.getTextContent());
-        messageForResponseBody.put("SentTs", message.getSentTs());
-        messageForResponseBody.put("ReceivedTs", message.getReceivedTs());
+    /**
+     * Renders the given parameters in a Map to return a successful HTTP response for all client requests to the ListMessages API.
+     */
+    public static Map<String, List<Map<String, Object>>> getSuccessResponseForListMessages(long userId, List<Message> messages) {
 
-        return messageForResponseBody;
+        List<Map<String, Object>> listOfMessages = new ArrayList<Map<String, Object>>();
+        
+        //Sorts the messages in  ascending order of Creation Timestamp
+        Collections.sort(messages, Comparator.comparing(Message::getCreationTs));
```

Comment:
This assumes that the given list is mutable and updates that.
we shouldn't mutate the given arguments.
Rather, you can created a sorted copy of it.

Reply:
Done